File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/CommunityContentReport/BaseSelectCommunityReportDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.CommunityContentReport
{
    public abstract class BaseSelectCommunityReportDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public string CommunityId { get; set; } = null!;
        public SelectCommunityReportCommunityDTO? Community { get; set; }
        public string ReporterId { get; set; } = null!;
        public SelectCommunityReportProfileDTO? Reporter { get; set; }
        public string ReportedProfileId { get; set; } = null!;
        public SelectCommunityReportProfileDTO? ReportedProfile { get; set; }
        public string Reason { get; set; } = null!;
        public ReportStatus Status { get; set; }
        public string? ResolvedById { get; set; }
        public SelectCommunityReportProfileDTO? ResolvedBy { get; set; }
        public string? ResolutionNotes { get; set; }
        public ReportResolutionAction ResolutionAction { get; set; }
        public DateTimeOffset? DateCreated { get; set; }
        public DateTimeOffset? DateModified { get; set; }
    }

    public class SelectCommunityReportProfileDTO
    {
        public string Id { get; set; } = null!;
        public string FullName { get; set; } = null!;
        public string? AvatarUrl { get; set; }
        public string? BackgroundUrl { get; set; }
        public string? Bio { get; set; }
        public int ReputationPoints { get; set; }
        public List<string> TechStacks { get; set; } = [];
        public bool IsPrivate { get; set; }
    }

    public class SelectCommunityReportCommunityDTO
    {
        public string Id { get; set; } = null!;
        public string Name { get; set; } = null!;
        public string Slug { get; set; } = null!;
        public string? CommunityCoverPhotoUrl { get; set; }
    }
}

File: frontend/src/types/community-content-report/base-select-community-report-dto.ts
import { ReportStatus } from "./report-status";
import { ReportResolutionAction } from "./report-resolution-action";

export interface SelectCommunityReportProfileDTO {
    id: string;
    fullName: string;
    avatarUrl?: string | null;
    backgroundUrl?: string | null;
    bio?: string | null;
    reputationPoints?: number;
    techStacks?: string[];
    isPrivate?: boolean;
}

export interface SelectCommunityReportCommunityDTO {
    id: string;
    name: string;
    slug: string;
    communityCoverPhotoUrl?: string | null;
}

export interface BaseSelectCommunityReportDTO {
    id: string;
    communityId: string;
    community?: SelectCommunityReportCommunityDTO | null;
    reporterId: string;
    reporter?: SelectCommunityReportProfileDTO | null;
    reportedProfileId: string;
    reportedProfile?: SelectCommunityReportProfileDTO | null;
    reason: string;
    status: ReportStatus;
    resolvedById?: string | null;
    resolvedBy?: SelectCommunityReportProfileDTO | null;
    resolutionNotes?: string | null;
    resolutionAction: ReportResolutionAction;
    dateCreated?: string | null;
    dateModified?: string | null;
}
